boot/freeldr/freeldr/arch/i386/pcmem.c

```diff
@@ -286,6 +290,22 @@ PcMemGetBiosMemoryMap(PFREELDR_MEMORY_DESCRIPTOR MemoryMap, ULONG MaxMemoryMapSi
             break;
         }
 
+        /* Extra (input) safety. */
+        if (Regs.x.ecx < 20 || BiosMemoryMapSize < Regs.x.ecx)
+        {
+            ERR("BIOS returned an invalid entry length! (would-be-PcBiosMapCount = %lu, Entry length = (20 <=) %lu (<= %lu))\n\n",
+                PcBiosMapCount, Regs.x.ecx, BiosMemoryMapSize);
+            if (PcBiosMapCount > 0 || BiosMemoryMapSize < Regs.x.ecx)
```

Comment:
I'm not sure I follow this logic. The if seems to say we're okay if there are no previous entries, and the current entry is too small. Can you explain why we want to assert in this case, but not others?

Reply:
I added a comment.
Ftr, a BSoD would seem overkill (wrt this first step) for a release user, the ASSERT is just to make sure a debug user notices the ERR.